# Dataset analysis

#### Load dependencies

In [6]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

### Raw data overview
Read the data from file and print the first entries. Every row represents an interaction between 2 people.

In [12]:
df = pd.read_csv('./data/raw_data.csv', encoding = "ISO-8859-1")
df.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  ...  \
0    1  1.0       0    1       1     1     10         7       NaN      4  ...   
1    1  1.0       0    1       1     1     10         7       NaN      3  ...   
2    1  1.0       0    1       1     1     10         7       NaN     10  ...   
3    1  1.0       0    1       1     1     10         7       NaN      5  ...   
4    1  1.0       0    1       1     1     10         7       NaN      7  ...   

   attr3_3  sinc3_3  intel3_3  fun3_3  amb3_3  attr5_3  sinc5_3  intel5_3  \
0      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
1      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
2      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
3      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
4      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   

   fun5_3  amb5_3  
0     NaN     NaN  
1     NaN     NaN  
2     NaN     NaN  
3     NaN     NaN  
4     NaN     NaN  

[5 rows x 195 columns]

Since the Dataset contains **195** columns, unfortunately it is not possible to have a meaningful overview on them just using the `head()` method.

Show details about columns and the contained values.

In [8]:
with pd.option_context('display.max_rows', None): # Remove the limit on displayed rows
    display(df.describe().T)

count        mean         std    min     25%     50%      75%  \
iid       8378.0  283.675937  158.583367   1.00  154.00  281.00  407.000   
id        8377.0    8.960248    5.491329   1.00    4.00    8.00   13.000   
gender    8378.0    0.500597    0.500029   0.00    0.00    1.00    1.000   
idg       8378.0   17.327166   10.940735   1.00    8.00   16.00   26.000   
condtn    8378.0    1.828837    0.376673   1.00    2.00    2.00    2.000   
wave      8378.0   11.350919    5.995903   1.00    7.00   11.00   15.000   
round     8378.0   16.872046    4.358458   5.00   14.00   18.00   20.000   
position  8378.0    9.042731    5.514939   1.00    4.00    8.00   13.000   
positin1  6532.0    9.295775    5.650199   1.00    4.00    9.00   14.000   
order     8378.0    8.927668    5.477009   1.00    4.00    8.00   13.000   
partner   8378.0    8.963595    5.491068   1.00    4.00    8.00   13.000   
pid       8368.0  283.863767  158.584899   1.00  154.00  281.00  408.000   
match     8378.0    0.164717    0.370947   0.00    0.00    0.00    0.000   
int_corr  8220.0    0.196010    0.303539  -0.83   -0.02    0.21    0.430   
samerace  8378.0    0.395799    0.489051   0.00    0.00    0.00    1.000   
age_o     8274.0   26.364999    3.563648  18.00   24.00   26.00   28.000   
race_o    8305.0    2.756653    1.230689   1.00    2.00    2.00    4.000   
pf_o_att  8289.0   22.495347   12.569802   0.00   15.00   20.00   25.000   
pf_o_sin  8289.0   17.396867    7.044003   0.00   15.00   18.37   20.000   
pf_o_int  8289.0   20.270759    6.782895   0.00   17.39   20.00   23.810   
pf_o_fun  8280.0   17.459714    6.085526   0.00   15.00   18.00   20.000   
pf_o_amb  8271.0   10.685375    6.126544   0.00    5.00   10.00   15.000   
pf_o_sha  8249.0   11.845930    6.362746   0.00    9.52   10.64   16.000   
dec_o     8378.0    0.419551    0.493515   0.00    0.00    0.00    1.000   
attr_o    8166.0    6.190411    1.950305   0.00    5.00    6.00    8.000   
sinc_o    8091.0    7.175256    1.740575   0.00    6.00    7.00    8.000   
intel_o   8072.0    7.369301    1.550501   0.00    6.00    7.00    8.000   
fun_o     8018.0    6.400599    1.954078   0.00    5.00    7.00    8.000   
amb_o     7656.0    6.778409    1.794080   0.00    6.00    7.00    8.000   
shar_o    7302.0    5.474870    2.156163   0.00    4.00    6.00    7.000   
like_o    8128.0    6.134498    1.841258   0.00    5.00    6.00    7.000   
prob_o    8060.0    5.208251    2.129354   0.00    4.00    5.00    7.000   
met_o     7993.0    1.960215    0.245925   1.00    2.00    2.00    2.000   
age       8283.0   26.358928    3.566763  18.00   24.00   26.00   28.000   
field_cd  8296.0    7.662488    3.758935   1.00    5.00    8.00   10.000   
race      8315.0    2.757186    1.230905   1.00    2.00    2.00    4.000   
imprace   8299.0    3.784793    2.845708   0.00    1.00    3.00    6.000   
imprelig  8299.0    3.651645    2.805237   1.00    1.00    3.00    6.000   
goal      8299.0    2.122063    1.407181   1.00    1.00    2.00    2.000   
date      8281.0    5.006762    1.444531   1.00    4.00    5.00    6.000   
go_out    8299.0    2.158091    1.105246   1.00    1.00    2.00    3.000   
career_c  8240.0    5.277791    3.309520   1.00    2.00    6.00    7.000   
sports    8299.0    6.425232    2.619024   1.00    4.00    7.00    9.000   
tvsports  8299.0    4.575491    2.801874   1.00    2.00    4.00    7.000   
exercise  8299.0    6.245813    2.418858   1.00    5.00    6.00    8.000   
dining    8299.0    7.783829    1.754868   1.00    7.00    8.00    9.000   
museums   8299.0    6.985781    2.052232   0.00    6.00    7.00    9.000   
art       8299.0    6.714544    2.263407   0.00    5.00    7.00    8.000   
hiking    8299.0    5.737077    2.570207   0.00    4.00    6.00    8.000   
gaming    8299.0    3.881191    2.620507   0.00    2.00    3.00    6.000   
clubbing  8299.0    5.745993    2.502218   0.00    4.00    6.00    8.000   
reading   8299.0    7.678515    2.006565   1.00    7.00    

Show the number of records present in the dataset.

In [9]:
len(df)

8378

The collected informations for each interaction are numerous and not all are relevant in our work. So we need to perform some preprocessing on the data.

### Data pre-processing

The document [SpeedDatingSurveyAndDataKey.doc](./data/SpeedDatingSurveyAndDataKey.doc) contains the survey given during the event, and therefore the semantics of the columns present in the dataset.

Remove irrelevant fields for our work.

In [14]:
irrelevant_cols = [
    'iid', 'id', 'idg',             # identifiers of the person
    #TODO: what is condtn?
    'wave',                         # number of the experiment's wave
    'round',                        # number of people that met in wave
    'position', 'positin1', 'order',# details about the logistics of the event
    'partner', 'pid',               # partner's identifiers
    #TODO what is pf_o_att
    'undergra', 'mn_sat', 'tuition',# details of the university career of the person
    'from', 'zipcode', 'income',    # details of the person's origins
    #'goal',                         # person's goal in participating in the experiment, in fact could be turned in goal in using the app, DISCUSS
    'exphappy', 'expnum',           # expectations on the event DISCUSS
    'dec', 'dec_o',                 # want to meet the person again
    'attr', 'sinc', 'intel', 'fun', # person's evaluation of the partner and of the meeting
    'amb', 'shar', 'like', 'prob',
    'met', 'match_es',
    'attr1_s','sinc1_s','intel1_s', # questions asked in the middle of the event
    'fun1_s', 'amb1_s', 'shar1_s',
    'attr3_s','sinc3_s','intel3_s',
    'fun3_s', 'amb3_s',
    'satis_2', 'length', 'numdat_2',# person's evaluation of the event
    'attr7_2','sinc7_2','intel7_2', # questions asked after the event
    'fun7_2', 'amb7_2', 'shar7_2',
    'you_call','them_cal','date_3', # dates after the event
    'numdat_3', 'num_in_3',
    'attr7_3','sinc7_3','intel7_3', # questions asked after the event
    'fun7_3', 'amb7_3', 'shar7_3',
    'attr4_3','sinc4_3','intel4_3',
    'fun4_3', 'amb4_3', 'shar4_3',
    'attr2_3','sinc2_3','intel2_3',
    'fun2_3', 'amb2_3', 'shar2_3',
    'attr3_3','sinc3_3','intel3_3',
    'fun3_3', 'amb3_3',
    'attr5_3','sinc5_3','intel5_3',
    'fun5_3', 'amb5_3',
]
df = df.drop(irrelevant_cols, axis=1)

In [15]:
df

gender  condtn  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0          0       1      0      0.14         0   27.0     2.0      35.0   
1          0       1      0      0.54         0   22.0     2.0      60.0   
2          0       1      1      0.16         1   22.0     4.0      19.0   
3          0       1      1      0.61         0   23.0     2.0      30.0   
4          0       1      1      0.21         0   24.0     3.0      30.0   
...      ...     ...    ...       ...       ...    ...     ...       ...   
8373       1       2      0      0.64         0   26.0     3.0      10.0   
8374       1       2      0      0.71         0   24.0     6.0      50.0   
8375       1       2      0     -0.46         0   29.0     3.0      40.0   
8376       1       2      0      0.62         0   22.0     4.0      10.0   
8377       1       2      0      0.01         0   22.0     4.0      20.0   

      pf_o_sin  pf_o_int  ...  sinc5_2  intel5_2  fun5_2  amb5_2  attr1_3  \
0         20.0      20.0  ...      NaN       NaN     NaN     NaN     15.0   
1          0.0       0.0  ...      NaN       NaN     NaN     NaN     15.0   
2         18.0      19.0  ...      NaN       NaN     NaN     NaN     15.0   
3          5.0      15.0  ...      NaN       NaN     NaN     NaN     15.0   
4         10.0      20.0  ...      NaN       NaN     NaN     NaN     15.0   
...        ...       ...  ...      ...       ...     ...     ...      ...   
8373      10.0      30.0  ...      3.0       9.0     4.0     7.0     70.0   
8374      20.0      10.0  ...      3.0       9.0     4.0     7.0     70.0   
8375      10.0      30.0  ...      3.0       9.0     4.0     7.0     70.0   
8376      25.0      25.0  ...      3.0       9.0     4.0     7.0     70.0   
8377      20.0      10.0  ...      3.0       9.0     4.0     7.0     70.0   

      sinc1_3  intel1_3  fun1_3  amb1_3  shar1_3  
0        20.0      20.0    15.0    15.0     15.0  
1        20.0      20.0    15.0    15.0     15.0  
2        20.0      20.0    15.0    15.0     15.0  
3        20.0      20.0    15.0    15.0     15.0  
4        20.0      20.0    15.0    15.0     15.0  
...       ...       ...     ...     ...      ...  
8373      0.0      20.0    10.0     0.0      0.0  
8374      0.0      20.0    10.0     0.0      0.0  
8375      0.0      20.0    10.0     0.0      0.0  
8376      0.0      20.0    10.0     0.0      0.0  
8377      0.0      20.0    10.0     0.0      0.0  

[8378 rows x 112 columns]